llvm/test/CodeGen/AMDGPU/lds-no-realign-allocated-variables.ll

```diff
@@ -0,0 +1,15 @@
+; RUN: opt -S -mtriple=amdgcn-- -amdgpu-lower-module-lds < %s | FileCheck %s
+
+; Can't have a second variable without absolute_symbol showing it is realigned as
+; there is a fatal error on mixing absolute and non-absolute symbols
+
+; CHECK: @lds.dont_realign = internal addrspace(3) global i64 undef, align 2, !absolute_symbol !0
+@lds.dont_realign = internal addrspace(3) global i64 undef, align 2, !absolute_symbol !0
```

Comment:
Use poison

Reply:
Why poison? Working model is that uses of uninitialised LDS are fine but meaningless, undef seems to capture that more accurately than poison

Reply:
Can we have one test with poison and one with undef? I've seen poison being used for these LDS vars in certain cases.

Reply:
It doesn't do much harm here but in general duplicating test cases for zero changes in code path seems bad

I'd lean towards the default for tests should probably correspond to whatever clang marks them with, which seems to be poison now but I'm sure used to be undef

Reply:
poison is new-ish so yes, it used to be undef

Reply:
I'm ok with just a test with poison

Reply:
Poison takes it for the test case, but in general it seems unfortunate. Reading from uninitialised LDS could be totally well defined and making it poison will propagate dead code elimination from it, though I concede that is the direction LLVM is moving in